### ISSUE TITLE
[Bug] Lyrics are not being downloaded

### ISSUE THREAD

**Issue by WildeBeast2521** [zotify-dev/zotify#196]
# Bug
Tried on many songs, with the '♪' or not, and they all give the same error:

```text
Spotify API Error (unknown): received an empty response
Skipping lyrics for [Artist - Song]: lyrics not available
```

The lyrics are not downloaded at all.

# Solution
The line provided in https://github.com/zotify-dev/zotify/issues/120 seems to work, by adding it to the `get_auth_header` function in `zotify.py`:
```python
'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36'
```
Please make the necessary commit.

> [!NOTE]
The solution is currently working intermittently.

**Comment by invalidexistance** [comment]
didn't worked for me

**Comment by WildeBeast2521** [comment]
> didn't worked for me 

@invalidexistance Stopped working for me as well.

**Comment by Le0X8** [comment]
@invalidexistance @WildeBeast2521 looks like Spotify did some changes in the last few weeks, now lyrics ripping can't be done with this version of Zotify anymore :(

**Comment by iProBaum** [comment]
so is there a way to download lyrics now or not? cuz the title says (Solved), therefore im a bit confused, would appreciate an answer :)

**Comment by WildeBeast2521** [comment]
> so is there a way to download lyrics now or not? cuz the title says (Solved), therefore im a bit confused, would appreciate an answer :)

@iProBaum Sorry. Should have fixed that.

Well, I have been using https://github.com/fashni/MxLRC instead. The project doesn't seem to be maintained anymore but it has been working for a couple of months now. The development of Zotify also seems to have come to a stop so it's unlikely this issue or many of Zotify's issues will be fixed in the future.

**Comment by iProBaum** [comment]
> > so is there a way to download lyrics now or not? cuz the title says (Solved), therefore im a bit confused, would appreciate an answer :)
> 
> [@iProBaum](https://github.com/iProBaum) Sorry. Should have fixed that.
> 
> Well, I have been using https://github.com/fashni/MxLRC instead. The project doesn't seem to be maintained anymore but it has been working for a couple of months now. The development of Zotify also seems to have come to a stop so it's unlikely this issue or many of Zotify's issues will be fixed in the future.

No worries, thanks for updating :) 
I have a question regarding the MxLRC project - I dont have much experience with python and coding and all of that, but I looked through the project and it said that it downloads lrc files, and my question is is there a way so that the mp3 files itself have the lyrics? I once downloaded with the DeezLoader Bot through Telegram, because I just needed one song and I had no access to my pc and there the mp4 file itself had the lyrics and in the Samsung Music Player there was like a button where the lyrics were in sync with the song, and I wonder if its gonne be like that or if it just gives me a file with the lyrics and If so, how is that helpful? I hope its clear what I mean xD would appreciate an answer :)

**Comment by iProBaum** [comment]
> > > so is there a way to download lyrics now or not? cuz the title says (Solved), therefore im a bit confused, would appreciate an answer :)
> > 
> > 
> > [@iProBaum](https://github.com/iProBaum) Sorry. Should have fixed that.
> > Well, I have been using https://github.com/fashni/MxLRC instead. The project doesn't seem to be maintained anymore but it has been working for a couple of months now. The development of Zotify also seems to have come to a stop so it's unlikely this issue or many of Zotify's issues will be fixed in the future.
> 
> No worries, thanks for updating :) I have a question regarding the MxLRC project - I dont have much experience with python and coding and all of that, but I looked through the project and it said that it downloads lrc files, and my question is is there a way so that the mp3 files itself have the lyrics? I once downloaded with the DeezLoader Bot through Telegram, because I just needed one song and I had no access to my pc and there the mp4 file itself had the lyrics and in the Samsung Music Player there was like a button where the lyrics were in sync with the song, and I wonder if its gonne be like that or if it just gives me a file with the lyrics and If so, how is that helpful? I hope its clear what I mean xD would appreciate an answer :)

I did a little bit of research and I found the answer with the help of ChatGPT: I found that the MxLRC project downloads separate LRC files rather than embedding lyrics into the MP3 files. For the Samsung Music Player to display synchronized lyrics, the MP3 and its corresponding LRC file must have the exact same name and be located in the same folder. This way, the player can automatically recognize and display the external lyrics file. Thanks anyways :)

**Comment by WildeBeast2521** [comment]
> No worries, thanks for updating :) 
> I have a question regarding the MxLRC project - I dont have much experience with python and coding and all of that, but I looked through the project and it said that it downloads lrc files, and my question is is there a way so that the mp3 files itself have the lyrics? I once downloaded with the DeezLoader Bot through Telegram, because I just needed one song and I had no access to my pc and there the mp4 file itself had the lyrics and in the Samsung Music Player there was like a button where the lyrics were in sync with the song, and I wonder if its gonne be like that or if it just gives me a file with the lyrics and If so, how is that helpful? I hope its clear what I mean xD would appreciate an answer :)

Yeah, that file must have had embedded lyrics, which MxLRC doesn't do. It only downloads and saves the synced lyrics in `.lrc` files in the directory, as music players can use those when lyrics are not embedded into the song.

**Comment by iProBaum** [comment]
> > No worries, thanks for updating :)
> > I have a question regarding the MxLRC project - I dont have much experience with python and coding and all of that, but I looked through the project and it said that it downloads lrc files, and my question is is there a way so that the mp3 files itself have the lyrics? I once downloaded with the DeezLoader Bot through Telegram, because I just needed one song and I had no access to my pc and there the mp4 file itself had the lyrics and in the Samsung Music Player there was like a button where the lyrics were in sync with the song, and I wonder if its gonne be like that or if it just gives me a file with the lyrics and If so, how is that helpful? I hope its clear what I mean xD would appreciate an answer :)
> 
> Yeah, that file must have had embedded lyrics, which MxLRC doesn't do. It only downloads and saves the synced lyrics in `.lrc` files in the directory, as music players can use those when lyrics are not embedded into the song.

thank you

**Comment by shinji257** [comment]
This is working for me in my fork but I added it to `get_auth_header_and_params` as well right below it.  I don't know if that is why it works for me.  I had put it there the first try (by accident) and it didn't work at that spot but left it there when I added it above and it started to work after.

**Comment by WildeBeast2521** [comment]
> This is working for me in my fork but I added it to `get_auth_header_and_params` as well right below it. I don't know if that is why it works for me. I had put it there the first try and it didn't work at that spot but left it there when I added it above and it started to work after.

Seems to have started to work again, with the solution I provided at the top. Quite strange indeed.

I'm going to leave the solution above and close the issue since the repository is not being maintained anymore.
